### PR TITLE
[full-ci] Display share-info details in right sidebar

### PR DESCRIPTION
**Pull request by paulcod3**
## Description
See https://github.com/owncloud/web/issues/5735

## Related Issue

- Fixes https://github.com/owncloud/web/issues/5735

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests

## Checklist:

- [x] Code changes
- [x] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link>

### CODE REVIEW

**Comment by ownclouders** [comment]
Results for <strong>oC10SharingInternalUsersSharingIndicator</strong> https://drone.owncloud.com/owncloud/web/18597/28/1
The following scenarios passed on retry:
- webUISharingInternalUsersToRootSharingIndicator/shareWithUsers.feature:98

**Comment by ownclouders** [comment]
Results for <strong>oC10iPhone2</strong> https://drone.owncloud.com/owncloud/web/18597/46/1
The following scenarios passed on retry:
- webUITrashbinRestore/trashbinRestore.feature:14

**Comment by paulcod3** [comment]
@kulmann @pascalwengerter would be nice if someone could tell me if the implementation is right so far before I write unittest just in case something is terribly wrong.

**Comment by ownclouders** [comment]
Results for <strong>oC10SharingInternalUsersSharingIndicator</strong> https://drone.owncloud.com/owncloud/web/18598/28/1
The following scenarios passed on retry:
- webUISharingInternalUsersSharingIndicator/shareWithUsers.feature:154
- webUISharingInternalUsersToRootSharingIndicator/shareWithUsers.feature:98

File: packages/web-app-files/src/components/SideBar/Details/FileDetails.vue [hunk 1]
```diff
@@ -117,9 +145,45 @@ export default {
...
       return this.$gettext('Overview of the information about the selected file')
     },
+    shareDateLabel() {
+      return this.$gettext('Shared on:')
+    },
+    sharedViaLabel() {
+      return this.$gettext('Shared via:')
+    },
+    shareDateTooltip() {
+      const date = new Date(this.sharedTime * 1000)
```
Comment (kulmann) [inline packages/web-app-files/src/components/SideBar/Details/FileDetails.vue:155]: Please use `luxon` for `date-to-human-readable`-conversions.

**Review by kulmann — CHANGES_REQUESTED** [review]

**Comment by ownclouders** [comment]
Results for <strong>oC10SharingExternalRoot</strong> https://drone.owncloud.com/owncloud/web/18598/40/1
The following scenarios passed on retry:
- webUISharingExternalToRoot/federationSharing.feature:356

**Comment by ownclouders** [comment]
Results for <strong>oC10XGAPortrait2</strong> https://drone.owncloud.com/owncloud/web/18598/43/1
The following scenarios passed on retry:
- webUITrashbinRestore/trashbinRestore.feature:14

**Comment by ownclouders** [comment]
Results for <strong>oC10SharingExternal</strong> https://drone.owncloud.com/owncloud/web/18598/39/1
The following scenarios passed on retry:
- webUISharingExternal/federationSharing.feature:363

**Comment by ownclouders** [comment]
Results for <strong>oC10SharingExternalRoot</strong> https://drone.owncloud.com/owncloud/web/18602/40/1
The following scenarios passed on retry:
- webUISharingExternalToRoot/federationSharing.feature:356

**Review by pascalwengerter — DISMISSED** [review]
Code looks good so far already, minor improvement ideas

File: packages/web-app-files/src/components/SideBar/Details/FileDetails.vue [hunk 2]
```diff
@@ -210,10 +285,28 @@ export default {
   watch: {
     highlightedFile() {
       this.loadData()
+      this.refreshShareDetailsTree()
+    },
+    sharesTreeLoading(current) {
```
[the remark this answers is not in the document]
Comment (paulcod3) [reply packages/web-app-files/src/components/SideBar/Details/FileDetails.vue:286]: We will be refactoring the share trees load function to be awaitable in the future.

[the remark this answers is not in the document]
Comment (kulmann) [reply packages/web-app-files/src/components/SideBar/Details/FileDetails.vue:286]: Agreed in a call to keep it like it is and refactor it once we have the shares tree in a good state.

**Review by kulmann — APPROVED** [review]
LGTM 👍

**Review by kulmann — APPROVED** [review]
LGTM 👍

**Comment by ownclouders** [comment]
Results for <strong>oC10SharingInternalGroupsSharingIndicator</strong> https://drone.owncloud.com/owncloud/web/18940/24/1
The following scenarios passed on retry:
- webUISharingInternalGroupsSharingIndicator/shareWithGroups.feature:114

**Pull request merged**